packages/sanes-browser-extension/src/extension/background/model/persona/index.ts

```diff
@@ -264,6 +266,25 @@ export class Persona {
     return balancesPerAccount;
   }
 
+  public async getStarnames(): Promise<readonly string[]> {
+    let starnames: readonly BnsUsernameNft[] = [];
+
+    const bnsConnection = this.getBnsConnection();
+    const accounts = await this.accountManager.accounts();
+    const bnsIdentities = accounts
+      .flatMap(account => account.identities)
+      .filter(ident => ident.chainId === bnsConnection.chainId());
+
+    await Promise.all(
+      bnsIdentities.map(async bnsIdentity => {
+        const bnsAddress = bnsCodec.identityToAddress(bnsIdentity);
+        starnames = [...(await bnsConnection.getUsernames({ owner: bnsAddress }))];
```

Comment:
You re-assign the `starnames` all the time instead of adding to it. Try `starnames.push(...(await bnsConnection.getUsernames({ owner: bnsAddress }))`. Then you use `const starnames: BnsUsernameNft[] = []` , i.e. only one array but mutable.